ddtrace/bootstrap/sitecustomize.py

```diff
@@ -74,6 +63,39 @@ def update_patched_modules():
         EXTRA_PATCHED_MODULES[module] = asbool(should_patch)
 
 
+if PY2:
+    _unloaded_modules = []
+
+
+def cleanup_loaded_modules():
+    # Unload all the modules that we have imported, expect for the ddtrace one.
+    for m in list(_ for _ in sys.modules if _ not in MODULES_LOADED_AT_STARTUP):
```

Comment:
@P403n1x87 What's the minimum set of modules we could unload here to solve gunicorn brokenness? I think those are the only modules we should apply this fix to as a first pass. It's at least `threading` - which others?

Reply:
It's hard to say. The `threading` module imports other stuff, and we should perhaps unload that too. But the fact that we know we need to unload `threading` is internal knowledge from `gevent`, which is what we want to avoid here.

Reply:
@P403n1x87 It seems to me like we could frame this decision as a trade-off. On one hand we have minimizing the amount of technical debt incurred by depending on gevent internals, and on the other we have minimizing risk by patching specific modules. Within this framing, I tend to prefer minimizing risk at the cost of depending on gevent internals. That is, I think avoiding module cloning wherever possible reduces unintended consequences enough to be worth some dependence on gevent internals. There is also the possibility that we could expand the set of cloned modules in a future change after the minimum set has had some time in production to give us feedback about how it works.

What do you think about this framing? We could also talk about this on video with the squad. Thanks again for all of the detailed responses.

Reply:
After a chat with @Yun-Kim and @P403n1x87 this morning, we settled on keeping the change as-is and putting it behind a feature flag as described in https://github.com/DataDog/dd-trace-py/pull/4863#discussion_r1098792818.